pkg/proxy/backend/backend_conn_mgr.go

```diff
@@ -264,9 +273,19 @@ func (mgr *BackendConnManager) getBackendIO(cctx ConnContext, auth *Authenticato
 
 // ExecuteCmd forwards messages between the client and the backend.
 // If it finds that the session is ready for redirection, it migrates the session.
-func (mgr *BackendConnManager) ExecuteCmd(ctx context.Context, request []byte) error {
+func (mgr *BackendConnManager) ExecuteCmd(ctx context.Context, request []byte) (err error) {
+	defer func() {
+		if err != nil {
+			if errors.Is(err, ErrBackendConn) {
+				mgr.quitSource = SrcBackendConn
+			} else if !errors.Is(err, ErrClientConn) {
+				mgr.quitSource = SrcClientErr
+			}
```

Comment:
Very similar to code in `Connect`, maybe we should have `setQuitSourceByError(err)`